src/Schema/Builder.php

```diff
@@ -107,6 +109,48 @@ public function dropAllTables()
         }
     }
 
+    public function getTables()
+    {
+        $db = $this->connection->getMongoDB();
+        $collections = [];
+
+        foreach ($db->listCollections() as $collectionInfos) {
+            $stats = $db->selectCollection($collectionInfos->getName())->aggregate([
+                ['$collStats' => ['storageStats' => ['scale' => 1]]],
+                ['$project' => ['storageStats.totalSize' => 1]],
+            ])->toArray();
+
+            $collections[] = [
+                'name' => $collectionInfos->getName(),
+                'schema' => null,
+                'size' => $stats[0]?->storageStats?->totalSize ?? null,
+                'comment' => null,
+                'collation' => null,
+                'engine' => null,
+            ];
+        }
+
+        usort($collections, function ($a, $b) {
+            return $a['name'] <=> $b['name'];
+        });
+
+        return $collections;
+    }
+
+    public function getTableListing()
+    {
+        $db = $this->connection->getMongoDB();
+        $collections = [];
+
+        foreach ($db->listCollections() as $collectionInfos) {
+            $collections[] = $collectionInfos->getName();
+        }
```

Comment:
This should definitely only use `listCollectionNames`:
```suggestion
        $collections = $db->listCollectionNames();
```